broker-core/src/main/java/io/zeebe/broker/workflow/model/transformation/handler/ActivityHandler.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Zeebe Broker Core
+ * Copyright © 2017 camunda services GmbH (info@camunda.com)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package io.zeebe.broker.workflow.model.transformation.handler;
+
+import io.zeebe.broker.workflow.model.BpmnStep;
+import io.zeebe.broker.workflow.model.ExecutableFlowNode;
+import io.zeebe.broker.workflow.model.ExecutableWorkflow;
+import io.zeebe.broker.workflow.model.transformation.ModelElementTransformer;
+import io.zeebe.broker.workflow.model.transformation.TransformContext;
+import io.zeebe.model.bpmn.instance.Activity;
+import io.zeebe.protocol.intent.WorkflowInstanceIntent;
+
+public class ActivityHandler implements ModelElementTransformer<Activity> {
+
+  @Override
+  public Class<Activity> getType() {
+    return Activity.class;
+  }
+
+  @Override
+  public void transform(Activity element, TransformContext context) {
+    final ExecutableWorkflow workflow = context.getCurrentWorkflow();
+    final ExecutableFlowNode activity =
+        workflow.getElementById(element.getId(), ExecutableFlowNode.class);
+
+    bindLifecycle(context, activity);
+  }
+
+  private void bindLifecycle(TransformContext context, final ExecutableFlowNode activity) {
+    activity.bindLifecycleState(
```

Comment:
Do we really need the extra abstraction level (BpmnStep)? I mean wouldn't it be simpler to directly add here the corresponding handlers? Say we have an Interface BpmnStep and each class, which implements this interface defines an step. This means you can call handle on each BpmnStep subclass, I think it is no need for the extra Enumeration.